tests/BuildScriptGenerator.Tests/SemanticVersionResolverTest.cs

```diff
@@ -94,5 +95,21 @@ public void SpecificVersionSpecified()
             // Assert;
             Assert.Equal("1.2.3", returnedVersion);
         }
+
+        [Theory]
+        [InlineData("1", "1.2.3", int.MinValue)]
+        [InlineData("1.2", "1.2.3", -1)]
+        [InlineData("1.3.4", "1.2.3", 1)]
+        [InlineData("3.5", "2.1.1", 1)]
+        [InlineData("3.5", "4.1", -1)]
+        [InlineData("3.5", "4", int.MinValue)]
+        public void CompareVersionsTest(string providedVersion, string supportedVersion, int resultExpected)
```

Comment:
What is the pivot (scenario) that you are trying to test here? And I would put that pivot as part of the test name.

Reply:
I'm trying to compare versions as many scenarios as I could think of. I will try to add some comment or rename the test to be more meaningful